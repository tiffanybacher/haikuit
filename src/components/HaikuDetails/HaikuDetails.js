import React, { Component } from 'react';
import { connect } from 'react-redux';
import HaikuForm from '../HaikuForm/HaikuForm';

export class HaikuDetails extends Component {
  render() {
    const id = parseInt(this.props.match.params.id);
    const haiku = this.props.haikus.find(haiku => {
      return haiku.id === id;
    });

    return (
      <section className="HaikuDetails">
        <h2>Edit Your Haiku</h2>
        <HaikuForm {...haiku} />
      </section>
    );
  }
}

export const mapStateToProps = (state) => ({
  haikus: state.haikus
});

export default connect(mapStateToProps)(HaikuDetails);
