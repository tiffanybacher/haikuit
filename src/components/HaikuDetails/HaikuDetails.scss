.HaikuDetails {
  background: white;
  padding: 20px;

  h2 {
    margin-top: 0;
  }
}

@media only screen and (min-width: 768px) {
  .HaikuDetails {
    font-size: 1.1em;
    padding: 30px 50px;
  }
}

@media only screen and (min-width: 1000px) {
  .HaikuDetails {
    font-size: 1.2em;
    padding: 40px 75px;
  }
}

@media only screen and (min-width: 1200px) {
  .HaikuDetails {
    padding: 50px 100px;
  }
}