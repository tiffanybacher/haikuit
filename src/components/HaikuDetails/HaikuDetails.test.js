import React from 'react';
import { shallow } from 'enzyme';
import { HaikuDetails, mapStateToProps } from './HaikuDetails';

describe('HaikuDetails', () => {
  let mockMatch;
  let mockHaikus;
  let wrapper;

  beforeEach(() => {
    mockMatch = { params: { id: 1} };
    mockHaikus = [{ title: 'Title', id: 1 }];
    wrapper = shallow(
      <HaikuDetails 
        match={mockMatch}
        haikus={mockHaikus} />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('map state to props', () => {
    it('should map haikus to props', () => {
      const mockState = {
        haikus: [{title: 'title'}]
      };
      const result = mapStateToProps(mockState);
      const expected = { haikus: mockState.haikus };

      expect(result).toEqual(expected);
    });
  });
});