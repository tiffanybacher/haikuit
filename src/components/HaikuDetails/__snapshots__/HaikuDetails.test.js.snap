// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HaikuDetails should match snapshot 1`] = `
<section
  className="HaikuDetails"
>
  <h2>
    Edit Your Haiku
  </h2>
  <Connect(HaikuForm)
    id={1}
    title="Title"
  />
</section>
`;
